docs/data/sdks/android-kotlin/index.md

```diff
@@ -180,3 +180,112 @@ With an empty `userId` and a completely new `deviceId`, the current user would a
 ```kotlin
 amplitude.reset()
 ```
+
+## Amplitude SDK Plugin
+
+Plugins allow you to extend Amplitude SDK's behavior by, for example, modifying event properties (enrichment type) or sending to a third-party APIs (destination type). A plugin is an object with methods `setup()` and `execute()`.
+
+### Plugin.setup
+
+This method contains logic for preparing the plugin for use and has `amplitude` instance as a parameter. The expected return value is `null`. A typical use for this method, is to instantiate plugin dependencies. This method is called when the plugin is registered to the client via `amplitude.add()`.
+
+### Plugin.execute
+
+This method contains the logic for processing events and has `event` instance as parameter. If used as enrichment type plugin, the expected return value is the modified/enriched event; while if used as a destination type plugin, the expected return value is `null`. This method is called for each event, including Identify, GroupIdentify and Revenue events, that is instrumented using the client interface.
+
```

Comment:
Should we mention the DestinationPlugin specific methods somewhere? e.g. `track()`

Reply:
Add the following sentence under the Destination Plugin section.  Let me know what you feel. 
`In destination plugin, you are able to overwrite the track(), identify(), groupIdentify(), revenue(), flush() functions.`